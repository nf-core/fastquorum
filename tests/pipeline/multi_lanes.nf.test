nextflow_pipeline {

    name "Test workflow - Multiple Lanes"
    script "main.nf"
    tag "multi_lanes"
    tag "pipeline"

    test("Multiple Lanes - R&D mode") {

        when {
            params {
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/testdata/samplesheets/samplesheet.multi_lanes.csv'
                fasta  = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/references/chr17.fa'
                outdir = "$outputDir"
                mode   = "rd"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("multi_lanes_rd_software_versions") },
                { assert workflow.trace.succeeded().size() == 39 },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_one_lane/SRR6109255_one_lane.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_two_lanes/SRR6109255_two_lanes.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_three_lanes/SRR6109255_three_lanes.cons.filtered.bam").exists() },
                { assert snapshot(
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_one_lane/SRR6109255_one_lane.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_one_lane/SRR6109255_one_lane.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_two_lanes/SRR6109255_two_lanes.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_two_lanes/SRR6109255_two_lanes.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_three_lanes/SRR6109255_three_lanes.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_three_lanes/SRR6109255_three_lanes.duplex_seq_metrics.family_sizes.txt"),
                ).match("multi_lanes_rd") }
            )
        }

    }

    test("Multiple Lanes - high throughput mode") {

        when {
            params {
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/testdata/samplesheets/samplesheet.multi_lanes.csv'
                fasta  = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/references/chr17.fa'
                outdir = "$outputDir"
                mode   = "ht"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("multi_lanes_ht_software_versions") },
                { assert workflow.trace.succeeded().size() == 36 },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_one_lane/SRR6109255_one_lane.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_two_lanes/SRR6109255_two_lanes.cons.filtered.bam").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_three_lanes/SRR6109255_three_lanes.cons.filtered.bam").exists() },
                { assert snapshot(
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_one_lane/SRR6109255_one_lane.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_one_lane/SRR6109255_one_lane.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_two_lanes/SRR6109255_two_lanes.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_two_lanes/SRR6109255_two_lanes.duplex_seq_metrics.family_sizes.txt"),
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_three_lanes/SRR6109255_three_lanes.grouped-family-sizes.txt"),
                    path("$outputDir/metrics/duplex_seq/SRR6109255_three_lanes/SRR6109255_three_lanes.duplex_seq_metrics.family_sizes.txt"),
                ).match("multi_lanes_ht") }
            )
        }

    }

}
