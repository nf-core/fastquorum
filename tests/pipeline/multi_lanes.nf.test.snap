{
    "multi_lanes_rd_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.18-r1243-dirty, fgbio=2.4.0, samtools=1.21}, ALIGN_RAW_BAM={bwa=0.7.18-r1243-dirty, fgbio=2.4.0, samtools=1.21}, CALLDDUPLEXCONSENSUSREADS={fgbio=2.4.0}, COLLECTDUPLEXSEQMETRICS={fgbio=2.4.0}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.4.0}, FILTERCONSENSUSREADS={fgbio=2.4.0}, GROUPREADSBYUMI={fgbio=2.4.0}, MERGE_BAM={samtools=1.21}, Workflow={nf-core/fastquorum=v1.1.1dev}}"
         ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-13T16:23:06.347717"
    },
    "multi_lanes_ht": {
        "content": [
            "SRR6109255_one_lane.grouped-family-sizes.txt:md5,3e0cd02882c86b850145aa1f43d95f69",
            "SRR6109255_one_lane.duplex_seq_metrics.family_sizes.txt:md5,852fdad2e8d82fd95d3b173baa329d0d",
            "SRR6109255_two_lanes.grouped-family-sizes.txt:md5,6cd7b37a9b89395104f080fae658854d",
            "SRR6109255_two_lanes.duplex_seq_metrics.family_sizes.txt:md5,e3b8b55271eab9489775a9a9cfdb5f78",
            "SRR6109255_three_lanes.grouped-family-sizes.txt:md5,4c720d70e6c0bd8fe38cb3befc608982",
            "SRR6109255_three_lanes.duplex_seq_metrics.family_sizes.txt:md5,dc78b321e46046f1ee1e5b3a678b79a2"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T01:42:46.597128"
    },
    "multi_lanes_ht_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.18-r1243-dirty, fgbio=2.4.0, samtools=1.21}, ALIGN_RAW_BAM={bwa=0.7.18-r1243-dirty, fgbio=2.4.0, samtools=1.21}, CALLANDFILTERDUPLEXCONSENSUSREADS={fgbio=2.4.0}, COLLECTDUPLEXSEQMETRICS={fgbio=2.4.0}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.4.0}, GROUPREADSBYUMI={fgbio=2.4.0}, MERGE_BAM={samtools=1.21}, Workflow={nf-core/fastquorum=v1.1.1dev}}"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-13T16:26:31.69708"
    },
    "multi_lanes_rd": {
        "content": [
            "SRR6109255_one_lane.grouped-family-sizes.txt:md5,3e0cd02882c86b850145aa1f43d95f69",
            "SRR6109255_one_lane.duplex_seq_metrics.family_sizes.txt:md5,852fdad2e8d82fd95d3b173baa329d0d",
            "SRR6109255_two_lanes.grouped-family-sizes.txt:md5,6cd7b37a9b89395104f080fae658854d",
            "SRR6109255_two_lanes.duplex_seq_metrics.family_sizes.txt:md5,e3b8b55271eab9489775a9a9cfdb5f78",
            "SRR6109255_three_lanes.grouped-family-sizes.txt:md5,4c720d70e6c0bd8fe38cb3befc608982",
            "SRR6109255_three_lanes.duplex_seq_metrics.family_sizes.txt:md5,dc78b321e46046f1ee1e5b3a678b79a2"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T01:34:44.993842"
    }
}
